cloud-native-declarative-osi-principles.md

```diff
@@ -1,44 +1,44 @@
 
 _**L7**_ _- If an infrastructure element that is part of a CNF is_ _**virtual layer 1**, it will be immutable_
 
 Virtualized cloud native networking infrastructure components that are part of the physical layer 1, such as virtual network cards, should have configuration that is immutable.
 
 _**L8**_ _- If an infrastructure element that is part of a CNF is_ _**virtual layer 2 or higher**, it will be_ _**immutable and orchestrated.**_
 
-Virtual layer 2 and higher network functions, such as layer 2 MPLS VPNs, should be provisioned immutably.
+Virtual layer 2 and higher network functions, such as layer 2 MPLS VPNs, should be provisioned immutably. Configuration for said network services should be captured in a template, stored with an associated version and 'pushed' via the higher level orchestration construct in an atomic fashion.
 
 _**L8**_ _- If a CNF is_ _**virtual layer 2 or higher**, it will expose itself using_ _**service discovery.**_
 
 _**L9**_ _- If a CNF is_ _**virtual layer 1**, its_ _**provisioning**_ _will use a_ _**server template**_
 
-The infrastructure elements of the lowest level virtual underlay network \(e.g. networking components of a hypervisor that map to the physical components of its node\), should have its configuration baked into an image that is versioned and managed with an image managing system. Whatever configuration that is not on the image should be applied after the initial image is deployed before the infrastructure element is considered ready for use.
+The infrastructure elements of the lowest level virtual underlay network \(e.g. networking components of a hypervisor that map to the physical components of its node\), should have its configuration baked into an artifact that is versioned and managed with an artifact management system. Whatever configuration that is not on the image should be applied after the initial artifact is deployed \(Day 2\), via an orchestrated and versioned process, before the infrastructure element is considered ready for use.
 
 _**L10**_ _- If an infrastructure element that is part of a CNF is_ _**virtual layer 2, 3, or higher**, its_ _**deployment**_ _will be within a_ _**microservice**_ _and will be_ _**orchestrated.**_
 
```

Comment:
I'd also like to insert an L11 here:

_**L11**_ _- If an infrastructure element that is part of a CNF is_ _**virtual layer 2, 3, or higher**, its_ _**configuration**_ _will be _ _**versioned**_ _within a_ _**template**_ _and deployed via_ _**orchestration.**_